### PR TITLE
cli/smartcontract: provide manifest groups on deploy

### LINKED ISSUE #2098: Specify contract groups in CLI
Our CLI currently can't set `Groups` field in contract  manifest. Group is a pair of a public key and a signature of contract hash.
We can include public keys in config and pass accounts for signing in command arguments.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #2098 .

There are possible problems, but I haven't found a way for them to occur:
1. Sent manifest will differ from the one we read. This is ok, because contract hash calculation uses only manifest name, so `calc-hash` will continue to work.
2. One might want to have multiple groups in different wallets (for some multisig transactions). This is not supported.

### CODE REVIEW

**Comment by roman-khimov** [comment]
Maybe we can solve both problems by allowing to modify the manifest with some commands adding signatures? Manifest is produced by compiler initially, but it's a file, we can edit it in whatever fashion. These commands would be somewhat similar to `wallet sign`.

**Comment by fyrchik** [comment]
@roman-khimov while we can move this to a separate command, it will be less convenient that it seems:
1. We need to provide both NEF file and sender to calculate contract hash.
2. Because of (1) we need to know sender in advance (out of all it is the biggest usability issue).
3. Some verification will be required during the deployment.

So while it is flexible, I think it is much less pleasant to use. I is a rarely used feature, though.
Thoughts @AnnaShaleva ?

**Comment by roman-khimov** [comment]
> We need to provide both NEF file and sender to calculate contract hash.

That's just two parameters.

> Some verification will be required during the deployment.

It's done automatically by native contract, we do test execution of transaction anyway and this execution will just fail if there is anything wrong with manifested groups.

> It is a rarely used feature, though.

That's why I think we should cover _all_ use cases right from the beginning. And have proper manifest file available to the user, because otherwise you'll be changing it somewhat behind the scenes.

**Comment by AnnaShaleva** [comment]
>  I think it is much less pleasant to use

I agree with that, it's much simpler to provide groups on deploy by using flags, but at the same time:
1. I think that the user should be able to see the resulting manifest file (the one that was sent to RPC node). Because if something goes wrong while deploying, the user should be aware of the whole set of the initial data.
2. That is more important, we should support adding several groups using different wallets. I don't think that this feature will stay unused.

All in all, I vote for Roman's solution: 

> These commands would be somewhat similar to wallet sign.

File: cli/smartcontract/manifest.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		return cli.NewExitError(errNoWallet, 1)
+	}
+
+	w, err := wallet.NewWalletFromFile(walletPath)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	defer w.Close()
+
+	addr, err := flags.ParseAddress(ctx.String("account"))
```
Comment (AnnaShaleva) [inline cli/smartcontract/manifest.go:30]: Let's use an address flag instead of a string?

File: cli/smartcontract/manifest.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	if err != nil {
+		return cli.NewExitError(errors.New("address is missing"), 1)
+	}
+
+	sender, err := flags.ParseAddress(ctx.String("sender"))
```
Comment (AnnaShaleva) [inline cli/smartcontract/manifest.go:35]: And for the sender.

File: cli/smartcontract/manifest.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	nefFile, err := nef.FileFromBytes(f)
+	if err != nil {
+		return nil, nil, err
```
Comment (AnnaShaleva) [inline cli/smartcontract/manifest.go:100]: Maybe we can return a more specific error, something like "invalid nef format"?

File: cli/smartcontract/manifest.go (hunk 1 above)
Comment (fyrchik) [reply cli/smartcontract/manifest.go:30]: I find it more verbose in case we require non-empty flags.

**Review by roman-khimov — COMMENTED** [review]
Documentation update?

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
Lint error should be fixed though.

**Review by AnnaShaleva — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**